pkg/execution/executor/executor.go

```diff
@@ -597,19 +597,29 @@ func (e *executor) Schedule(ctx context.Context, req execution.ScheduleRequest)
 		}
 	}
 
-	err := e.smv2.Create(ctx, newState)
+	st, err := e.smv2.Create(ctx, newState)
 	switch err {
-	case nil:
-		// no-op, continue
-	case state.ErrIdentifierExists:
-		return nil, err
+	case nil, state.ErrIdentifierExists:
+		// no-op: this is safe, continue
 	default:
 		return nil, fmt.Errorf("error creating run state: %w", err)
 	}
+	// override the runID from the state if it doesn't match.
+	// this can happen on scheduling retries due to errors like networking, or abrupt failures
+	//
+	// TODO: update the code from here on to reference the `state` returned instead of using
+	// previous values to avoid weird mismatch issues.
+	// right now, the only mismatch should just be the runID
+	if metadata.ID.RunID != st.RunID() {
+		metadata.ID.RunID = st.RunID()
+	}
 
 	//
 	// Create cancellation pauses immediately, only if this is a non-batch event.
 	//
+	// NOTE: on scheduling retries, this will result in duplicated pauses for cancellation.
```

Comment:
Or should we create a deterministic pause ID to only create a single pause?

Reply:
https://github.com/inngest/inngest/blob/8eef881818dc6e9fd6c63b83146779c1800ed1a4/pkg/execution/executor/executor.go#L2525

Reply:
https://github.com/inngest/inngest/blob/8eef881818dc6e9fd6c63b83146779c1800ed1a4/pkg/execution/executor/executor.go#L2525